NeoPixelBus Light breaks logger

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):** 
pip on macOS

**ESP (ESP32/ESP8266, Board/Sonoff):** 
Wemos D1 Mini

**Affected component:** 
[Neopixelbus Light](https://esphome.io/components/light/neopixelbus.html)

**Description of problem:**
Including a Neopixelbus Light and also using logger in an automation won't compile.

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: wemos_button
  platform: ESP8266
  board: d1_mini

logger:

binary_sensor:
  - platform: gpio
    pin:
      number: GPIO0
      inverted: True
    name: "button"
    on_click:
      min_length: 50ms
      max_length: 350ms
      then:
        - logger.log: "Click"

light:
  - platform: neopixelbus
    type: GRB
    pin: GPIO3
    num_leds: 1
    name: "indicator"
```

**Traceback (if applicable):**
<!--
Please copy the traceback here if compilation is failing. If possible, also connect to the ESP over USB and copy its logs into the backticks.
-->
```
Compiling .pioenvs/wemos_button/src/main.cpp.o
In file included from /Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/Arduino.h:259:0,
from .piolibdeps/esphome-core/src/esphome/espmath.h:4,
from .piolibdeps/esphome-core/src/esphome/automation.h:5,
from .piolibdeps/esphome-core/src/esphome/application.h:7,
from .piolibdeps/esphome-core/src/esphome.h:1,
from src/main.cpp:3:
/Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/pgmspace.h:16:51: error: __c causes a section type conflict with __c
#define PSTR(s) (__extension__({static const char __c[] PROGMEM = (s); &__c[0];}))
^
/Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/WString.h:38:76: note: in definition of macro 'FPSTR'
#define FPSTR(pstr_pointer) (reinterpret_cast<const __FlashStringHelper *>(pstr_pointer))
^
/Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/WString.h:39:34: note: in expansion of macro 'PSTR'
#define F(string_literal) (FPSTR(PSTR(string_literal)))
^
.piolibdeps/esphome-core/src/esphome/log.h:80:3: note: in expansion of macro 'F'
F(ESPHOME_LOG_COLOR_##letter "[" #letter "][%s:%03u]: " format ESPHOME_LOG_RESET_COLOR), tag, __LINE__
^
.piolibdeps/esphome-core/src/esphome/log.h:106:49: note: in expansion of macro 'ESPHOME_LOG_FORMAT'
esp_log_printf_(ESPHOME_LOG_LEVEL_DEBUG, tag, ESPHOME_LOG_FORMAT(tag, D, format), ##__VA_ARGS__)
^
.piolibdeps/esphome-core/src/esphome/log.h:168:28: note: in expansion of macro 'esph_log_d'
#define ESP_LOGD(tag, ...) esph_log_d(tag, __VA_ARGS__)
^
src/main.cpp:21:7: note: in expansion of macro 'ESP_LOGD'
ESP_LOGD("main", "Click");
^
/Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/pgmspace.h:16:51: note: '__c' was declared here
#define PSTR(s) (__extension__({static const char __c[] PROGMEM = (s); &__c[0];}))
^
/Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/WString.h:38:76: note: in definition of macro 'FPSTR'
#define FPSTR(pstr_pointer) (reinterpret_cast<const __FlashStringHelper *>(pstr_pointer))
^
/Users/Matthew/.platformio/packages/framework-arduinoespressif8266@2.20402.4/cores/esp8266/WString.h:39:34: note: in expansion of macro 'PSTR'
#define F(string_literal) (FPSTR(PSTR(string_literal)))
^
.piolibdeps/esphome-core/src/esphome/log.h:80:3: note: in expansion of macro 'F'
F(ESPHOME_LOG_COLOR_##letter "[" #letter "][%s:%03u]: " format ESPHOME_LOG_RESET_COLOR), tag, __LINE__
^
.piolibdeps/esphome-core/src/esphome/log.h:109:49: note: in expansion of macro 'ESPHOME_LOG_FORMAT'
esp_log_printf_(ESPHOME_LOG_LEVEL_DEBUG, tag, ESPHOME_LOG_FORMAT(tag, C, format), ##__VA_ARGS__)
^
.piolibdeps/esphome-core/src/esphome/log.h:170:33: note: in expansion of macro 'esph_log_config'
#define ESP_LOGCONFIG(tag, ...) esph_log_config(tag, __VA_ARGS__)
^
.piolibdeps/esphome-core/src/esphome/light/neo_pixel_bus_light_output.tcc:64:3: note: in expansion of macro 'ESP_LOGCONFIG'
ESP_LOGCONFIG(TAG, "  Num LEDs: %u", this->controller_->PixelCount());
^
```

**Additional information and things you've tried:**




<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/src/esphome/light/neo_pixel_bus_light_output.tcc
+++ b/src/esphome/light/neo_pixel_bus_light_output.tcc
@@ -50,7 +50,6 @@ void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::add_leds(
 }
 template<typename T_METHOD, typename T_COLOR_FEATURE>
 void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::setup() {
-  ESP_LOGCONFIG(TAG, "Setting up NeoPixelBus light...");
   for (int i = 0; i < this->size(); i++) {
     (*this)[i] = ESPColor(0, 0, 0, 0);
   }
@@ -60,8 +59,6 @@ void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::setup() {
 }
 template<typename T_METHOD, typename T_COLOR_FEATURE>
 void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::dump_config() {
-  ESP_LOGCONFIG(TAG, "NeoPixelBus light:");
-  ESP_LOGCONFIG(TAG, "  Num LEDs: %u", this->controller_->PixelCount());
 }
 template<typename T_METHOD, typename T_COLOR_FEATURE>
 void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::loop() {
@@ -92,7 +89,6 @@ void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::loop() {
   }
 #endif
 
-  ESP_LOGVV(TAG, "Writing RGB values to bus...");
   this->controller_->Show();
 }
 template<typename T_METHOD, typename T_COLOR_FEATURE>

--- a/src/esphome/light/neo_pixel_bus_light_output.tcc
+++ b/src/esphome/light/neo_pixel_bus_light_output.tcc
@@ -58,8 +58,7 @@ void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::setup() {
   this->controller_->Begin();
 }
 template<typename T_METHOD, typename T_COLOR_FEATURE>
-void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::dump_config() {
-}
+void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::dump_config() {}
 template<typename T_METHOD, typename T_COLOR_FEATURE>
 void NeoPixelBusLightOutputBase<T_METHOD, T_COLOR_FEATURE>::loop() {
   if (!this->should_show_())

